cfme/tests/infrastructure/test_providers.py

```diff
@@ -492,3 +495,54 @@ def test_infra_discovery_screen(appliance):
                    "to_ip4": ips['to']})
         view.start.click()
         view.flash.assert_message(ips['msg'])
+
+
+@pytest.fixture
+def setup_provider_min_templates(request, appliance, provider, min_templates):
+    templates_yaml = len(provider.data.get('templates', {}))
+    if templates_yaml < min_templates:
+        pytest.skip(f'Number of templates defined in yaml for {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    if len(provider.mgmt.list_templates()) < min_templates:
+        pytest.skip(f'Number of templates on {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    # Function-scoped fixture to set up a provider
+    setup_or_skip(request, provider)
+
+
+def verify_checked_items_compared(checkedList, view):
+    for e in view.comparison_table.headers[1:-1]:
```

Comment:
Please clean up these magic numbers, indexing, and using single character vars.

What is `e`? standing for here?

What are you indexing out of the headers? Let's comment in-line and name it more descriptively.

Below, what's the function of the split and zero-index? Again let's descriptively name what this value is.

Why use try/except around a remove, instead of directly asserting that the item from the checked list is in the list of header values?

I think you could use two `set` objects to verify the list of headers and list of items that had been selected are the same.

Consider having this as a method in the `ComparisonView` class, instead of just a function within the test module.  That way its re-usable across the various entities in MIQ that can be compared, and can be used outside of this test module.

Reply:
I'll make the vars more descriptive, add some inline comments. I could put in a function like get_templates_on_page() and put the 'magic' in there. This is all a result of what is contained in .headers. The values returned, except for the fitst and last are the templates in the view. The first and last items are just part of the table. So I used [1:-1] on these. I'll add an inline comment. Then the template-like names have additional characters added to them so I split on space and take the first([0]) item. I'll make this clearer.

I was using try/except because it enabled me to distinguish between missing and extra templates in the view and would stop as soon as a failure was found. But "set" might be a good solution also. I'll look into it.

I agree with putting this functionality into methods for the view classes, and am working on this now in another PR. I am starting with implementing comparison for hosts while investigating what would be needed to extend to templates and datastores. I'd really like to get this test merged and running while I work on the optimization.

The changes are forthcoming, but I wanted to continue the discussion. Thanks.

Reply:
I pushed some changes but realize we have more discussion and changes to come.

Reply:
I take it you are suggesting creating sets from the 2 lists and taking the symetric difference. If It's non-zero we have a failure. Only thing is, it takes more steps to determine which lists have the extra/missing items. Or are you suggesting something else?